opencog/cython/opencog/bindlink.pyx

```diff
@@ -1,15 +1,22 @@
-from opencog.atomspace cimport Atom, AtomSpace, TruthValue
-from opencog.atomspace cimport cHandle, cAtomSpace, cTruthValue
-from opencog.atomspace cimport tv_ptr, strength_t, count_t
+from opencog.atomspace cimport Atom, AtomSpace, TruthValue, PtrHolder
+from opencog.atomspace cimport cAtomSpace, cTruthValue
+from opencog.atomspace cimport tv_ptr, strength_t, count_t, shared_ptr
 from opencog.atomspace cimport handle_cast
 from cython.operator cimport dereference as deref
 
+from opencog.atomspace import is_a, types, create_value_by_type
+
 def execute_atom(AtomSpace atomspace, Atom atom):
     if atom == None: raise ValueError("execute_atom atom is: None")
     cdef cValuePtr c_value_ptr = c_execute_atom(atomspace.atomspace,
                                            deref(atom.handle))
-    cdef cHandle c_result = handle_cast(c_value_ptr)
-    return Atom.createAtom(c_result, atomspace)
+    type = deref(c_value_ptr).get_type()
```

Comment:
Following three lines are not needed, `create_value_by_type` will construct `Atom` instance if required

Reply:
Updated.